frontend/src/app/patients/PatientForm.tsx

```diff
@@ -441,6 +446,34 @@ const PatientForm = (props: Props) => {
       props.saveCallback ? props.saveCallback() : patientInfoConfirmRedirect();
     }
   }
+
+  const onRaceChange = (race: Race) => {
```

Comment:
This is all a bit of a mess, mostly because it seems `patient` has no type in this form. I don't know if it's too out-of-scope for this PR, but it'd be great to actually have a type for patient and then just have one change handler (that still enforces type safety!) for all fields:

```typescript
const onPatientChange = <K extends keyof PatientType>(field: K) => (value: PatientType[K]) => {
  setFormChanged(true);
  setPatient({ ...patient, [field]: value });
}
```

And then use in each input would be (for example):

```jsx
<RadioGroup
  legend="Ethnicity"
  name="ethnicity"
  buttons={ETHNICITY_VALUES}
  selectedRadio={patient.ethnicity}
  onChange={onChange}
  onChange={onPatientChange("ethnicity")}
/>
```

Reply:
This is a great idea. As you mentioned the main reason I am avoiding this is that there is no type for Patient. I am going to submit a refactor of the entire `PatientForm.tsx` before adding the address validation in. I will include this change in that refactor